src/js/https_client.js

```diff
@@ -0,0 +1,156 @@
+/* Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var util = require('util');
+var incoming = require('https_incoming');
+var stream = require('stream');
+var Buffer = require('buffer');
+var httpsNative = process.binding(process.binding.https);
+
+var methods = {'0': 'DELETE', '1': 'GET', '2': 'HEAD', '3': 'POST',
+    '4': 'PUT', '5': 'CONNECT', '6': 'OPTIONS', '7': 'TRACE'};
+exports.METHODS = methods;
+
+function ClientRequest(options, cb) {
+  this.stream = stream.Writable.call(this, options);
+
+  // get port, host and method.
+  var port = options.port = options.port || 443;
+  var host = options.host = options.hostname || options.host || '127.0.0.1';
+  var path = options.path || '/';
+  var protocol = options.protocol || 'https:';
+
+  this.host = protocol + '//' + host + ':' + port + path;
+  this.method = options.method || 'GET';
+  this.ca = options.ca || '';
+  this.cert = options.cert || '';
+  this.key = options.key || '';
+
+  var isMethodGood = false;
+  for (var key in methods) {
+    if (methods.hasOwnProperty(key)) {
+      if(this.method == methods[key]) {
```

Comment:
Done. I did not check manually because I assumed `tools/check-tidy.py` and the `clang-format` will take care of things like this.